src/ocaml-output/Makefile

```diff
@@ -157,13 +161,13 @@ COMMIT = $(shell ../tools/get_commit)
 
 .PHONY: FStar_Version.ml
 FStar_Version.ml:
-	echo 'open FStar_Util' > $@
-	echo 'let dummy () = ();;' >> $@
-	echo 'FStar_Options._version := "$(VERSION)";;' >> $@
-	echo 'FStar_Options._platform := "$(PLATFORM)";;' >> $@
-	echo 'FStar_Options._compiler := "$(COMPILER)";;' >> $@
-	echo 'FStar_Options._date := "$(DATE)";;' >> $@
-	echo 'FStar_Options._commit:= "$(COMMIT)";;' >> $@
+	echo 'open FStar_Util' > $@.tmp
+	echo 'let dummy () = ();;' >> $@.tmp
+	echo 'FStar_Options._version := "$(VERSION)";;' >> $@.tmp
+	echo 'FStar_Options._platform := "$(PLATFORM)";;' >> $@.tmp
+	echo 'FStar_Options._compiler := "$(COMPILER)";;' >> $@.tmp
+	echo 'FStar_Options._commit:= "$(COMMIT)";;' >> $@.tmp
+	./copy_if_diff.sh $@.tmp $@
```

Comment:
you should be able to get rid of that copy_if_diff, too, since ocamlbuild is content-hash based...?

Reply:
yeah, you're right. Nice!

Reply:
yeah, that works. although unconditionally writing to FStar_Version.ml causes its mtime to exceed that of fstar.exe, because ocamlbuild is idemptotent based on hashes, no work is actually done.